okhttp/src/main/java/okhttp3/ResponseBody.java

```diff
@@ -103,7 +103,10 @@ public final Reader charStream() {
   /**
    * Returns the response as a string decoded with the charset of the Content-Type header. If that
    * header is either absent or lacks a charset, this will attempt to decode the response body as
-   * UTF-8.
+   * UTF-8. Closes {@link ResponseBody} automatically.
+   *
+   * <p>Warning: this method loads whole response body into memory and may cause
```

Comment:
The warning is extremely scary, and doesn’t mention that it will only trigger an OutOfMemoryError if the response is very large. I’m worried about people not using this method when they should because of this scary warning.